app/src/main/admin/actions/adminTouchstoneActionCreators.ts

```diff
@@ -30,6 +39,36 @@ export const adminTouchstoneActionCreators = {
         }
     },
 
+    setCurrentTouchstoneResponsibility(responsibility: AnnotatedResponsibility) {
+        return {
+            type: TouchstoneTypes.SET_CURRENT_TOUCHSTONE_RESPONSIBILITY,
+            data: responsibility
+        };
+    },
+
+    getResponsibilityCommentsForTouchstoneVersion(touchstoneVersion: string) {
+        return async (dispatch: Dispatch<AdminAppState>, getState: () => AdminAppState) => {
+                const responsibilityCommentSets: ResponsibilitySetWithComments[] = await (new TouchstonesService(dispatch, getState))
+                        .getResponsibilityCommentsForTouchstoneVersion(touchstoneVersion);
+                dispatch({
+                        type: TouchstoneTypes.RESPONSIBILITY_COMMENTS_FOR_TOUCHSTONE_VERSION_FETCHED,
+                        data: responsibilityCommentSets
+                } as ResponsibilityCommentsForTouchstoneVersionFetched);
+            }
+    },
+
+    addResponsibilityComment(touchstoneVersion: string, modellingGroupId: string, scenarioId: string, comment: string) {
+        return async (dispatch: Dispatch<AdminAppState>, getState: () => AdminAppState) => {
+            const service = new TouchstonesService(dispatch, getState);
+            const result = await service.addResponsibilityComment(touchstoneVersion, modellingGroupId, scenarioId, comment);
+            if (result) {
+                dispatch(this.setCurrentTouchstoneResponsibility(null));
```

Comment:
I suppose there's an argument that setting current touchstone responsibility back to null could be initiated directly by the modal on save as it is in cancel, to decouple these actions from each other - but I don't mind too much!